Firestore/core/src/core/expressions_eval.h

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_CORE_EXPRESSIONS_EVAL_H_
+#define FIRESTORE_CORE_SRC_CORE_EXPRESSIONS_EVAL_H_
+
+#include <memory>
+#include <utility>
+
+#include "Firestore/Protos/nanopb/google/firestore/v1/document.nanopb.h"
+#include "Firestore/core/src/api/expressions.h"
+#include "Firestore/core/src/api/stages.h"
+#include "Firestore/core/src/nanopb/message.h"
+
+namespace firebase {
+namespace firestore {
+namespace core {
+
+/** Represents the result of evaluating an expression. */
+class EvaluateResult {
+ public:
+  enum class ResultType {
```

Comment:
Presumably we'll need to add the Extended Types here in the future? :'-)

Reply:
Yeah...